imports/api/custom-users.js

```diff
@@ -40,6 +42,19 @@ Meteor.methods({
     })
     if (!invitedUser) throw new Meteor.Error('The code is invalid or login is required first')
 
+    // Track accesses
+    AccessInvitations.upsert({
+      userId: invitedUser._id
+    }, {
+      $set: {
+        userId: invitedUser._id
+      },
+      $push: {
```

Comment:
Why are you pushing dates and caseIds separately?
You can push a single object that bundles each caseId with the date it was added in

Reply:
I still think this data could be denormalized a bit so each combination of user Id and case Id can be stored in a separate document, but if you think we won't have a lot of other uses for this, then I'm ok with it

Reply:
Sorry, I just think it's easier on my brain and I don't really know how to use aggregate to make it work your way.

Reply:
Yeah, it's not so intuitive to use, but it should be something like:
```.aggregate({
  $group: {
    _id: { userId: '$userId' },
    cases: { $push: '$caseId' },
    firstOccurance: { $first: '$creationDate' }
  }
})```

Reply:
So you want me to change it ?

Reply:
I think it would enable a wider range of usages of this data for analytics in the future, but I'm not sure if it'll be necessary

Reply:
I'd prefer this being as it is. Sorry, I don't see the use case since I think all the useful data is there.

I mean, this was designed as a lightweight collection for tracking invites and I think it does it.

Reply:
Use case: Query invites by use case would become easier with your design suggestion.

Reply:
Actually we should query by "unitId" since users can't be invited to a case if they have already been invited to another case in product / property.